hystrix/circuit.go

```diff
@@ -67,13 +68,13 @@ func Flush() {
 
 // newCircuitBreaker creates a CircuitBreaker with associated Health
 func newCircuitBreaker(name string) *CircuitBreaker {
-	c := &CircuitBreaker{}
-	c.Name = name
-	c.metrics = newMetricExchange(name)
-	c.executorPool = newExecutorPool(name)
-	c.mutex = &sync.RWMutex{}
-
-	return c
+	return &CircuitBreaker{
+		Name:         name,
+		metrics:      newMetricExchange(name),
+		executorPool: newExecutorPool(name),
+		mutex:        &sync.RWMutex{},
+		logger:       getSettings(name).Logger,
```

Comment:
This could panic when getSettings returns nil

Reply:
It looks like getSettings does not return nil. If it can't find circuit settings for the named circuit, it configures one then returns those settings.

```
func getSettings(name string) *Settings {
	settingsMutex.RLock()
	s, exists := circuitSettings[name]
	settingsMutex.RUnlock()

	if !exists {
		ConfigureCommand(name, CommandConfig{})
		s = getSettings(name)
	}

	return s
}
``